cJSON.c

```diff
@@ -882,10 +889,22 @@ char *cJSON_PrintBuffered(const cJSON *item, int prebuffer, cjbool fmt)
     }
     p.length = prebuffer;
     p.offset = 0;
+    p.noalloc = 0;
 
     return print_value(item, 0, fmt, &p);
 }
 
+int cJSON_PrintPreallocated(cJSON *item,char *buf, const size_t len, const cjbool fmt)
+{
+    char *out;
+    printbuffer p;
+    p.buffer = buf;
+    p.length = len;
+    p.offset = 0;
+    p.noalloc = 1;
+    out = print_value(item,0,fmt,&p);
+    return (out != buf ? -1 : 0);
```

Comment:
This should return `out` like the other print functions. It will be `NULL` if it failed.